khmer/utils.py

```diff
@@ -260,9 +260,16 @@ def write_record_pair(read1, read2, fileobj):
         fileobj.write(recstr)
 
 
-def clean_input_reads(screed_iter):
-    for record in screed_iter:
-        record.cleaned_seq = record.sequence.upper().replace('N', 'A')
+def clean_input_reads(records):
+    """Add a cleaned_seq attribute to records that do not have one
+
+    Use this to convert a stream of records that might not have a
+    `cleaned_seq` attribute to one that does. Use this to extend
+    Records loaded by `screed.open()`.
+    """
+    for record in records:
+        if not hasattr(record, 'cleaned_seq'):
```

Comment:
It depends. Without this if you accidentally pass a stream of `khmer.Read`s into this then it will break because all attributes are readonly. So mostly a policy question.